### PR TITLE
RFE: Multi IPFailover router installations support on a single cluster

### PR DESCRIPTION
**Pull request by ramr**
Adds option and support for router id offset. Address customer RFE to enable multiple ipfailover router installations to run within the same cluster.

@smarterclayton  PTAL Thx 

@codificat  FYI

### CODE REVIEW

**Comment by codificat** [comment]
Thanks @ramr 
For reference, the RFE where this comes from is [Bug 1300298](https://bugzilla.redhat.com/show_bug.cgi?id=1300298).

**Comment by ramr** [comment]
@knobunc @marun FYI

**Comment by eparis** [comment]
[test]
@marun can you review please?

File: contrib/completions/bash/oadm [hunk 1]
```diff
@@ -2160,6 +2160,7 @@ _oadm_ipfailover()
     two_word_flags+=("-t")
     flags+=("--type=")
     flags+=("--virtual-ips=")
+    flags+=("--vrrpid-offset=")
```
Comment (marun) [inline contrib/completions/bash/oadm]: Out of curiosity, are these completions generated or manually curated?

Comment (eparis) [inline contrib/completions/bash/oadm]: generated. I wrote golang code to generate bash code   :)

**Comment by marun** [comment]
It appears possible for a user to choose offsets that would result in conflicting vrrpid's, e.g the following contrived example:

 - instance one uses 30 with 2 vips.
 - instance two uses 31 with 2 vips

Is it worth validating that the range of vrrpid's for all instances will not conflict?

File: pkg/cmd/experimental/ipfailover/ipfailover.go [hunk 2]
```diff
@@ -80,6 +81,7 @@ func NewCmdIPFailoverConfig(f *clientcmd.Factory, parentName, name string, out i
 	cmd.Flags().StringVarP(&options.NetworkInterface, "interface", "i", "", "Network interface bound by VRRP to use for the set of virtual IP ranges/addresses specified.")
 
 	cmd.Flags().IntVarP(&options.WatchPort, "watch-port", "w", ipfailover.DefaultWatchPort, "Port to monitor or watch for resource availability.")
+	cmd.Flags().IntVar(&options.VrrpIdOffset, "vrrpid-offset", options.VrrpIdOffset, "Offset to use for setting ids of VRRP instances (default offset is 0). This allows multiple ipfailover instances to be run within the same cluster.")
```
Comment (marun) [inline pkg/cmd/experimental/ipfailover/ipfailover.go]: nitiest of nits: Consider changing "to be run" => "to run"

File: contrib/completions/bash/oadm (hunk 1 above)
Comment (smarterclayton) [inline contrib/completions/bash/oadm]: Probably better than the other way around.

File: pkg/cmd/experimental/ipfailover/ipfailover.go (hunk 2 above)
Comment (smarterclayton) [inline pkg/cmd/experimental/ipfailover/ipfailover.go]: `--vrrp-id-offset`, `VRRPIDOffset" internally

**Comment by smarterclayton** [comment]
A few nits, otherwise LG

**Comment by ramr** [comment]
rebased + had to generate conversions (unrelated for some reason)  and made the changes as per review comments. PTAL  thx 

@marun as re: the validation for overlapping VRRP id offsets, we could potentially do that statically at creation/command time (we would have to store + query other related ipfailover objects - the offset values + the range/replicas - a wee bit complicated there) but even with that there's other ways to get
into a similar state with offsets that overlap because of say the number of replicas changing due to scaling [auto or manual].  Feels like there would be little bang for buck plus we still won't be able to address all cases and not to mention the added complexity. In a way, its not any different than say the case of specifying conflicting IPs.
IMHO, better to leave that onus on the user.

**Comment by smarterclayton** [comment]
Changes LGTM, go ahead and squash (jenkins is broke tonight).

**Comment by smarterclayton** [comment]
[merge] once Jenkins comes back from the grave

**Pull request merged**